iothub/device/src/HsmAuthentication/SasTokenBuilder.cs

```diff
@@ -36,12 +36,7 @@ public static string BuildExpiresOn(DateTime startTime, TimeSpan timeToLive)
 
         public static string BuildAudience(string iotHub, string deviceId, string moduleId)
         {
-            string audience = WebUtility.UrlEncode("{0}/devices/{1}/modules/{2}".FormatInvariant(
-                iotHub,
-                deviceId,
-                moduleId));
-
-            return audience;
+            return WebUtility.UrlEncode($"{iotHub}/devices/{WebUtility.UrlEncode(deviceId)}/modules/{WebUtility.UrlEncode(moduleId)}");
```

Comment:
@varunpuranik I don't think this is correct. I believe the "/" characters should not be escaped.

Reply:
This looks correct - IoThub will UrlDecode the audience (and then UrlDecode the deviceId and moduleId)